microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/utils/EnvUtils.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.utils;
+
+/**
+ * Environment variable utility class
+ *
+ */
+public class EnvUtils {
+	/**
+	 * Return true if the variable is an ENV variable, and false otherwise
+	 *
+	 * @param variable the variable to check
+	 * @return true if the variable is an ENV variable, and false otherwise
+	 */
+	public static boolean isEnvVariable(String variable) {
+		char[] chars = variable.toCharArray();
+		for (char c : chars) {
+			if (Character.isLetter(c)) {
+				return Character.isUpperCase(c);
```

Comment:
So then `Somevalue` is an environment variable according to this logic, because we decide on the first value only.

You can determine something is not an environment variable if some character fails the test, and "short circuit" the later checks, but to confirm it is an environment variable, you would need to check every letter-based character.